### ISSUE TITLE
[RFE] Ignore skipped inspection results, if no security problems were detected

### ISSUE THREAD

**Issue by msrb** [rpminspect/rpminspect#1409]
**Is your feature request related to a problem? Please describe.**

Inspections that contain security checks cannot be skipped by maintainers. This works as expected. However, let's consider the following scenario:

- a maintainer attempts to disable the `removedfiles` inspection in their local `rpminspect.yaml` file
- rpminspect force-runs the inspection as `removedfiles` is considered to be security-related
- rpminspect detects a potential problem in the `removedfiles` inspection, but this problem is not security-related
- `removedfiles` inspection is marked as failed and the build is blocked in CI/gating

**Describe the solution you'd like**

If a security-related inspection is disabled in the local config, but rpminspect later force-runs it, the inspection should fail ("bad" result), only if security-related problems were detected.
If all findings were non-security related, then then the result should be set to something like "ignored".

**Describe alternatives you've considered**

In theory, CI could be doing post-processing on rpminspect results, but I think it would be better to have this implemented directly in rpminspect.

**Additional context**

This was reported by kernel maintainers. They hit this problem in RHEL CI recently. See OSCI-7075 for more information.

**Comment by dcantrell** [comment]
Can you point me to two kernel builds that demonstrate what you are explaining?  I think this is a reasonable enhancement.

**Comment by ckelleyRH** [comment]
> Can you point me to two kernel builds that demonstrate what you are explaining? I think this is a reasonable enhancement.

This was discovered during gating for `kernel-6.10.0-0.rc2.10.el10`

**Comment by msrb** [comment]
Also here (OSCI-7075):

```
=> rpminspect.yaml contents:
+ cat rpminspect.yaml
# additional rpminspect configuration for this branch

---
inspections:
    abidiff: off
    kmidiff: off
    upstream: off
    subpackages: off
    license: off
    debuginfo: off
    removedfiles: off

badfuncs:
    ignore:
        - /usr/libexec/ksamples/*
        - /usr/libexec/kselftests/*

emptyrpm:
    expected_empty:
        - kernel
        - kernel-debug
        - kernel-debug-devel-matched
        - kernel-devel-matched
        - kernel-lpae
        - kernel-zfcpdump
        - kernel-zfcpdump-devel-matched
        - kernel-zfcpdump-modules

patches:
    ignore_list:
        - linux-kernel-test.patch
        - patch-5.14-redhat.patch
        - patch-%{patchversion}-redhat.patch

runpath:
    ignore:
        - /usr/libexec/kselftests/bpf/urandom_read
        - /usr/libexec/kselftests/bpf/no_alu32/urandom_read

debuginfo:
    ignore:
        - /usr/libexec/kselftests/bpf/*
        - /usr/lib/debug/usr/libexec/perf-core/tests/shell/coresight/*

elf:
    ignore:
        - /usr/libexec/perf-core/tests/shell/coresight/*
        - /usr/lib/debug/usr/libexec/perf-core/tests/shell/coresight/*

$ rpminspect-redhat --verbose --release=el9 --profile=rhel-9 kernel-5.14.0-457.el9 kernel-5.14.0-459.el9
```

Results (failed removedfiles inspection) are [here](https://artifacts.osci.redhat.com/testing-farm/0691dcbe-5a5c-4f13-8983-9cee42f4c5bb/work-rpminspectm2i5gvwp/rpminspect/execute/data/guest/default-0/rpminspect-1/data/viewer.html#).

**Comment by dcantrell** [comment]
I have been able to reproduce this locally.  Looks like what the inspection was incorrectly doing is setting the severity to RESULT_VERIFY if it was not a rebase comparison and it did not a corresponding security rule definition or security path prefix.  So that meant comparing a 5.x to 6.x kernel job would report removedfiles findings as INFO but comparing two 5.x builds would show removedfiles findings as VERIFY even though there was no security rule or matching path prefix.

Patch coming soon.
